pep-0536.txt

```diff
@@ -0,0 +1,184 @@
+syntax highlighter able to correctly highlight code making use of this.
+
+As a consequence, the expression portions would be harder to recognize
+with and without the aid of syntax highlighting.  With the new grammar,
+it is easy to extend syntax highlighters to correctly parse
+and display f-literals:
+
+.. raw:: html
+
+   <pre><span style=color:#ff5500>f'Magic wand: </span><span style=color:#3daee9>{</span>bag[<span style=color:#bf0303>'wand'</span>]<span style=color:#3daee9>:^10}</span><span style=color:#ff5500>'</span></pre>
+
+.. This is the output of kate-syntax-highlighter when given that code
+   (with some quotes stripped)
+
+Highlighting expression portions with possible escape sequences would
+mean to create a modified copy of all rules of the complete expression
+grammar, accounting for the possibility of escape sequences in key words,
+delimiters, and all other language syntax. One such duplication would
+yield one level of escaping depth and have to be repeated for a deeper
+escaping in a recursive f-literal. This is the case since no highlighting
+engine known to the author supports expanding escape sequences before
+applying rules to a certain context. Nesting contexts however is a
+standard feature of all highlighting engines.
+
+Familiarity also plays a role: Arbitrary nesting of expressions
+without expansion of escape sequences is available in every single
+other language employing a string interpolation method that uses
+expressions instead of just variable names. [#]_
+
+Specification
+=============
+
+PEP 498 specified f-strings as the following, but places restrictions on it::
+
+    f ' <text> { <expression> <optional !s, !r, or !a> <optional : format specifier> } <text> ... '
+
+All restrictions mentioned in the PEP are lifted from f-literals,
+as explained below:
+
+#. Expression portions may now contain strings delimited with the same
+   kind of quote that is used to delimit the f-literal.
+#. Backslashes may now appear within expressions just like anywhere else
+   in Python code.  In case of strings nested within f-literals,
+   escape sequences are expanded when the innermost string is evaluated.
+#. Comments, using the ``'#'`` character, are possible only in multi-line
+   f-literals, since comments are terminated by the end of the line
+   (which makes closing a single-line f-literal impossible).
+#. Expression portions may contain ``':'`` or ``'!'`` wherever
+   syntactically valid.  The first ``':'`` or ``'!'`` that is not part
+   of an expression has to be followed a valid coercion or format specifier.
+
+A remaining restriction not explicitly mentioned by PEP 498 is line breaks
+in expression portions.  Since strings delimited by single ``'`` or ``"``
+characters are expected to be single line, line breaks remain illegal
+in expression portions of single line strings.
+
+.. note:: Is lifting of the restrictions sufficient,
+   or should we specify a more complete grammar?
```

Comment:
I don't know of a specific one off the top of my head. You're going to have to do this for the implementation anyway and so you can just embed the diff to the grammar file.